src/classes/frFundraisingEventRegistration.cls

```diff
@@ -62,25 +60,36 @@ public class frFundraisingEventRegistration {
         String funraiseEventId = String.valueOf(request.get('eventId'));
         List<Fundraising_Event__c> event = [SELECT Id from Fundraising_Event__c WHERE fr_Id__c = :funraiseEventId];
         if(event.isEmpty()) {
-            insert new Error__c(Error__c =
-                                'Failed to find related record: Event with id ' + funraiseEventId+ ' for registration id ' + funraiseId);
+            frUtil.logRelationshipError(frUtil.Entity.REGISTRATION, funraiseId, 
+                                        frUtil.Entity.EVENT, funraiseEventId);
             return;
         }
         
         String funraiseSupporterId = String.valueOf(request.get('supporterId'));
         List<Contact> supporter = [SELECT Id from Contact WHERE fr_Id__c = :funraiseSupporterId];
         if(supporter.isEmpty()) {
-            insert new Error__c(Error__c =
-                                'Failed to find related record: Supporter with id ' + funraiseSupporterId+ ' for registration id ' + funraiseId);
+            frUtil.logRelationshipError(frUtil.Entity.REGISTRATION, funraiseId, 
+                                        frUtil.Entity.SUPPORTER, funraiseSupporterId,
+                                       'Registration supporter, not guest of');
             return;
         }
         String funraiseGuestOfId = String.valueOf(request.get('guestOfId'));
         List<Contact> guestOf = [SELECT Id from Contact WHERE fr_Id__c = :funraiseGuestOfId];
-        if(guestOf.isEmpty() && funraiseGuestOfId != null) {
+        if(guestOf.isEmpty() && String.isNotBlank(funraiseGuestOfId)) {
             //if it wasn't found, log an error but continue
-            insert new Error__c(Error__c =
-                                'Failed to find related record: Guest Of supporter with id ' + funraiseGuestOfId+ ' for registration id ' + funraiseId);
+            frUtil.logRelationshipError(frUtil.Entity.REGISTRATION, funraiseId, 
+                                        frUtil.Entity.SUPPORTER, funraiseSupporterId,
+                                        'Registration guest of, not supporter');
         }
+        
+        String funraiseTransactionId = String.valueOf(request.get('transactionId'));
+        List<Opportunity> transactions = [SELECT Id FROM Opportunity WHERE fr_Id__c = :funraiseTransactionId];
```

Comment:
There is a bug here where we are querying before we check if the value is nonnull which will lead to connecting the registration to any opportunity with a null fr_Id__c.  We need to make sure it's not blank before querying

Reply:
Ok pushed a commit to fix this and added tests to confirm the behavior as well